### PR TITLE
Update and add some new icons

### PR DESCRIPTION
**Pull request by jameskoster**
## Description

* "Cloud" icon updated to match g2 visual design
* Add "Mail" icon
* Add "Warning" icon

Screen Shot 2020-06-15 at 4 37 31 PM

Design discussion occurred [here](https://github.com/WordPress/gutenberg/issues/21995). I changed "alert" to "warning" for alignment with the [`Notice`](https://github.com/WordPress/gutenberg/tree/wp/5.4/packages/components/src/notice) component.

Also updates the wordpress/primitives dependency path in the Icons package.

I also had to update the path to the `wordpress/primitives` dependency in the Icons package.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. 

Closes #22699.

### CODE REVIEW

File: packages/icons/src/library/cloud.js [hunk 1]
```diff
@@ -4,8 +4,8 @@
 import { SVG, Path } from '@wordpress/primitives';
 
 const cloud = (
-	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="-2 -2 24 24">
```
Comment (senadir) [inline packages/icons/src/library/cloud.js:7]: `-2 -2` was meant to offset some icons that are designed on the 20px grid, assuming the new icon isn't.

**Review by senadir — APPROVED** [review]

File: packages/icons/src/library/cloud.js (hunk 1 above)
Comment (jameskoster) [reply packages/icons/src/library/cloud.js:7]: correct, all the new "g2" icons are on a 24px grid.

**Comment by jameskoster** [comment]
It doesn't seem to be related to the changes here, but I'm trying to update the snapshots for the e2e tests that are failing in Travis anyway. Either way, I'm just getting errors 🙈

If an engineer has a moment to look into this I'd really appreciate it!

**Comment by senadir** [comment]
I will look into it

**Comment by senadir** [comment]
the trick is to restart the tests until the stars align for you.

**Comment by senadir** [comment]
until a conflict happen because another PR was merged and now have to do it again

**Comment by aduth** [comment]
Last time icons were updated, there were end-to-end snapshots which needed to be updated. Hard to tell if that's the case here, since the build is (at the time of writing) in progress.

**Comment by senadir** [comment]
the tests passed before the conflict, the issue was caused by a selector timing out, resetting the test worked.

**Comment by senadir** [comment]
It seems like tests are not running?

**Comment by aduth** [comment]
They may have been queued. It appears to be running now.

**Comment by Garance91540** [comment]
Hi @aduth ! It seems like there's still an issue with the tests, would you be able to take a look and merge this is everything is ok? Thank you!

**Comment by mcsf** [comment]
👋 
Is the [illustration](https://user-images.githubusercontent.com/28450084/80712647-8b2e2f80-8af2-11ea-9050-087d81b51ef0.png) provided in the parent issue still the correct preview of the new icons? If so, could you edit the PR description to include that same image? It's always helpful in the future to be able to quickly trace these changes. Thanks!

**Comment by Garance91540** [comment]
I can't edit @jameskoster PR but here is the screenshot of the new icons

Screen Shot 2020-06-15 at 4 37 31 PM

**Comment by Garance91540** [comment]
Hi @senadir , would it be possible for you to review this PR?

**Comment by aristath** [comment]
Looks like this one needs to be rebased as there is currently a merge conflict.
Rebasing will most likely also fix the tests that got stuck...

**Comment by Mamaduka** [comment]
Hi, @jameskoster

Can you rebase this PR one more time?

**Pull request closed without merging**